modules/cluser-logging-troubleshooting-loki-entry-out-of-order-errors.adoc

```diff
@@ -0,0 +1,124 @@
+:_module-type: PROCEDURE
+
+[id="cluser-logging-troubleshooting-loki-entry-out-of-order-messages_{context}"]
+= Troubleshooting Loki "entry out of order" errors
+
+Forwarding a large block of messages to a Grafana Loki logging system exceeds the rate limit and generates "entry out of order" errors. To fix this issue, you update some values in the Loki server configuration file, `loki.yaml`.
+
+.Prerequisites
+
+* You have configured the `ClusterLogForwarder` custom resource to forward logs to Grafana Loki.
+
+* Your system sends a block of messages that is larger than 2 MB to Loki, such as:
++
+----
+"values":[["1630410392689800468","{\"kind\":\"Event\",\"apiVersion\":\
+.......
+......
+......
+......
+\"received_at\":\"2021-08-31T11:46:32.800278+00:00\",\"version\":\"1.7.4 1.6.0\"}},\"@timestamp\":\"2021-08-31T11:46:32.799692+00:00\",\"viaq_index_name\":\"audit-write\",\"viaq_msg_id\":\"MzFjYjJkZjItNjY0MC00YWU4LWIwMTEtNGNmM2E5ZmViMGU4\",\"log_type\":\"audit\"}"]]}]}
+----
+
+* When you enter `oc logs -c fluentd`, the Fluentd logs in your OpenShift Logging cluster show the following messages:
++
+[source,text]
+----
+429 Too Many Requests Ingestion rate limit exceeded (limit: 8388608 bytes/sec) while attempting to ingest '2140' lines totaling '3285284' bytes
+
+429 Too Many Requests Ingestion rate limit exceeded' or '500 Internal Server Error rpc error: code = ResourceExhausted desc = grpc: received message larger than max (5277702 vs. 4194304)'
+----
+
+* When you open the logs on the Loki server, they display `entry out of order` messages like these:
++
+[source,text]
+----
+,\nentry with timestamp 2021-08-18 05:58:55.061936 +0000 UTC ignored, reason: 'entry out of order' for stream:
+
+{fluentd_thread=\"flush_thread_0\", log_type=\"audit\"},\nentry with timestamp 2021-08-18 06:01:18.290229 +0000 UTC ignored, reason: 'entry out of order' for stream: {fluentd_thread="flush_thread_0", log_type="audit"}
+----
+
+.Procedure
+
+* Update the following fields in the Loki server configuration file, `loki.yaml`, with the values shown here:
+
+  * `grpc_server_max_recv_msg_size: 8388608`
+  * `chunk_target_size: 8388608`
+  * `ingestion_rate_mb: 8`
+  * `ingestion_burst_size_mb: 16`
+
+.Example `loki.yaml` file
```

Comment:
Makes sense for  loki servers managed by the user. Does this apply to Grafana hosted loki? @periklis is there a remedy there?

Reply:
https://grafana.com/docs/loki/latest/configuration/

Reply:
@alanconway I've asked @periklis to have a look at this question.

Reply:
@alanconway I don't know what knows Grafana Cloud offers to let users customize the managed Loki instances. IIRC they don't open that up at all to users.

Reply:
@periklis @alanconway If I understand correctly, it sounds like this troubleshooting topic is moot. Should I close this PR and Jira issue?

Reply:
I am not aware of the background/context of this issue. If the intent was to give folks a pointer to know where to start troubleshoot their Loki clusters, then it is ok. Otherwise any intent regarding Grafana renders this moot.